hazelcast/src/main/java/com/hazelcast/map/impl/recordstore/expiry/ExpirySystem.java

```diff
@@ -195,14 +203,36 @@ public final void extendExpiryTime(Data dataKey, long now) {
         }
 
         ExpiryMetadata expiryMetadata = getExpiryMetadataForExpiryCheck(dataKey, expireTimeByKey);
-        if (expiryMetadata == null
-                || expiryMetadata.getMaxIdle() == Long.MAX_VALUE) {
+        if (expiryMetadata == null) {
+            return;
+        }
+
+        long maxIdle = expiryMetadata.getMaxIdle();
+        if (maxIdle == Long.MAX_VALUE) {
+            return;
+        }
+
+        long ttl = expiryMetadata.getTtl();
+        if (ttl <= maxIdle) {
             return;
         }
 
-        long expirationTime = ExpirationTimeSetter.calculateExpirationTime(expiryMetadata.getTtl(),
-                expiryMetadata.getMaxIdle(), now);
-        expiryMetadata.setExpirationTime(expirationTime);
+        checkIfTtlGreaterThanMaxIdle(recordStore.getName(),
```

Comment:
I think it would be enough if we did this check only during the put accesses. I think it will never fail here since it will always fail in there before.